ysh/expr_parse.py

```diff
@@ -260,6 +260,32 @@ def _PushYshTokens(parse_ctx, gr, p, lex):
 
             continue
 
+        # @[  for array splice - convert expression to list of strings
+        if 0: #tok.id == Id.Left_AtLBracket:
```

Comment:
I removed this part because we only need Id.Expr_CastedDummy as a "hack" for things that are not expressions

Example:

```
var x = $(echo hi > out)    
```

There, the thing inside $() is not an expression.  So the parser generator `pgen2` can't handle it.  So we add a **hack** for the type system / homogeneous tree -- the entire command AST for the CommandSub is considered  **token** of type `Id.Expr_CastedDummy`, and then we cast it back to a `command_t` later

I'm just explaining why we have the Expr_CastedDummy, and why we don't need it in this case

---

Here, everything is an expression:

```
var x = @[42 + a[i]]
```

So we don't need to "hack up" the parse tree output from pgen2 (before turning it into an aST)